### PR TITLE
Add foldmason createdb

### LINKED ISSUE #7179: new module: foldmason/createdb
### Is there an existing module for this?

- [x] I have searched for the existing module

### Is there an open PR for this?

- [x] I have searched for existing PRs

### Is there an open issue for this?

- [x] I have searched for existing issues

### Further Information

add new module: foldmason/createdb

### Are you going to work on this?

- [ ] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by luisas**
Add a module for creating the database by Foldmason 

## PR checklist

Closes #7179 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/foldmason/createdb/main.nf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+process FOLDMASON_CREATEDB {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "community.wave.seqera.io/library/foldmason:2.7bd21ed--e7f739473ad6578d"
```
Comment (JoseEspinosa) [inline modules/nf-core/foldmason/createdb/main.nf]:
```suggestion
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/a8/a88d162c3f39a1518d48c3faec235e6fcde750586da868b62fc5f0a08a89aa9d/data' :
        'community.wave.seqera.io/library/foldmason:2.7bd21ed--e7f739473ad6578d' }"
```

File: modules/nf-core/foldmason/createdb/main.nf [hunk 2]
```diff
@@ -0,0 +1,47 @@
+process FOLDMASON_CREATEDB {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+            'oras://community.wave.seqera.io/library/foldmason:2.7bd21ed--a45f76ed12b391e6':
```
Comment (JoseEspinosa) [inline modules/nf-core/foldmason/createdb/main.nf:7]:
```suggestion
            'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/a8/a88d162c3f39a1518d48c3faec235e6fcde750586da868b62fc5f0a08a89aa9d/data' :
```

File: modules/nf-core/foldmason/createdb/main.nf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            'community.wave.seqera.io/library/foldmason:2.7bd21ed--e7f739473ad6578d' }"
+
+    input:
+    tuple val(meta) , path(pdbs)
```
Comment (JoseEspinosa) [inline modules/nf-core/foldmason/createdb/main.nf:11]: To make it generic for `pdb`s and `mmcif`s I will suggest to call it structures instead
```suggestion
    tuple val(meta) , path(structures)
```

File: modules/nf-core/foldmason/createdb/main.nf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    foldmason createdb \\
+        ${pdbs} \\
```
Comment (JoseEspinosa) [inline modules/nf-core/foldmason/createdb/main.nf:25]:
```suggestion
        ${structures} \\
```

File: modules/nf-core/foldmason/createdb/meta.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+input:
+  - - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. `[ id:'sample1', single_end:false ]`
+    - pdbs:
+        type: file
+        description: Input protein structures in PDB format.
```
Comment (JoseEspinosa) [inline modules/nf-core/foldmason/createdb/meta.yml:26]:
```suggestion
    - structures:
        type: file
        description: Input protein structures in `PDB` or `mmCIF` format.
```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Some small suggestions

File: modules/nf-core/foldmason/createdb/main.nf (hunk 2 above)
Comment (luisas) [reply modules/nf-core/foldmason/createdb/main.nf:7]: Why this one?

**Pull request merged**